hack/local-up-cluster.sh

```diff
@@ -140,10 +145,40 @@ else
     echo "skipped the build."
 fi
 
+
+function test_etcd {
+  etcd_path=$(which etcd)
+  if [ -z "$etcd_path" ]
+  then
+     echo
+     echo "Failed to find 'etcd' in your PATH, please fix and retry."
+     exit 1
+  fi
+}
+
 function test_docker {
     ${DOCKER[@]} ps 2> /dev/null 1> /dev/null
     if [ "$?" != "0" ]; then
-      echo "Failed to successfully run 'docker ps', please verify that docker is installed and \$DOCKER_HOST is set correctly."
+      echo
+      cat <<'EOF' >&2
+Failed to successfully run 'docker ps', please verify that docker is installed and \$DOCKER_HOST is set correctly. 
```

Comment:
they look pretty similar; it appears that the main difference is that `ensure_docker_daemon_connectivity` is using `docker info`, whereas this uses `docker ps`, but I think the effect is basically the same: if `docker` is installed and working, both should succeed; if `docker` isn't working, both will fail.